lnwallet/wallet.go

```diff
@@ -518,52 +553,60 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 // and change reserved for the channel, and derives the keys to use for this
 // channel.
 func (l *LightningWallet) initOurContribution(reservation *ChannelReservation,
-	selected *coinSelection, nodeAddr net.Addr, nodeID *btcec.PublicKey) error {
+	fundingIntent chanfunding.Intent, nodeAddr net.Addr,
+	nodeID *btcec.PublicKey, keyRing keychain.KeyRing) error {
 
 	// Grab the mutex on the ChannelReservation to ensure thread-safety
 	reservation.Lock()
 	defer reservation.Unlock()
 
-	if selected != nil {
-		reservation.ourContribution.Inputs = selected.coins
-		reservation.ourContribution.ChangeOutputs = selected.change
+	// At this point, if we have a funding intent, we'll use it to populate
+	// the existing reservation state entries for our coin selection.
+	if fundingIntent != nil {
+		if intent, ok := fundingIntent.(*chanfunding.FullIntent); ok {
+			for _, coin := range intent.InputCoins {
+				reservation.ourContribution.Inputs = append(
+					reservation.ourContribution.Inputs,
+					&wire.TxIn{
+						PreviousOutPoint: coin.OutPoint,
+					},
+				)
+			}
+			reservation.ourContribution.ChangeOutputs = intent.ChangeOutputs
+		}
+
+		reservation.fundingIntent = fundingIntent
```

Comment:
would it make sense to define a type of "remote funding intent"? Such that if you're the the responder in the workflow, then you also have an intent to refer to.

Reply:
Not sure if you've seen later, but we extend things to allow a caller to inject an intent. This is meant for the case when you're the responder to an extra-protocol initiated funding flow.